### ISSUE TITLE
Don't disable low-battery hibernation when setting HibernateDelaySec

### ISSUE THREAD

**Issue by boudekerk** [systemd/systemd#26498]
### Component

other

### Is your feature request related to a problem? Please describe

From the v253 release notes:
> If 'HibernateDelaySec=' is set to any value,
> the automated estimate (and thus the automated hibernation on low
> battery to avoid data loss) functionality will be disabled.

I don't understand why these must be mutually exclusive. To Hibernate to avoid data loss is an excellent feature, the usecase to actively want to disable that would be very rare I'd assume. Another great feature (for laptops) is the delayed hibernation via suspend-then-hibernate with HibernateDelaySec. It allows one to balance convenience with power saving.

### Describe the solution you'd like

A best of both worlds would be to hibernate on low battery, or after HibernateDelaySec. Whichever comes first.

### The systemd version you checked that didn't have the feature you are asking for

253

**Comment by YHNdnzj** [comment]
Actually I think this has been the case in the current version 🤔

@bluca Could you please point out where we disabled the functionality? Maybe I'm missing something...

**Comment by boudekerk** [comment]
If indeed it 'cannot happen', could that please be explained? Conceptually I don't understand why they would be.

Perhaps it's not possible in the current flow of the code, but in that case this issue would still be valid. (Although the gain might perhaps not warrant the effort needed to implement)

**Comment by OJFord** [comment]
To summarise:
- v251 & earlier used `HibernateDelaySec` to define how long to wait, suspended, before the hibernation part of `suspend-then-hibernate`;
- v252 changed it to define, as I understand, how long to wait, suspended, between periodic battery tests to see if we would be below 5% the next time, and should thus hibernate; (regression #25269)
- v253 restored the <=251 behaviour for users relying on it, moving the v252 behaviour to `SuspendEstimationSec` (PR #25374)

But I thought the intention of @yuwata's work there _was_ to support both. Certainly the diff to `man systemd-sleep.conf` suggests so (quoting the added description, my emphasis):
>  If the system supports low-battery alarms (ACPI _BTP), then the system will be woken up by
          the ACPI low-battery signal and hibernated (the state is then stored on disk). **Also**, if not
          interrupted within the timespan specified by <varname>HibernateDelaySec=</varname> **or** the estimated
          timespan until the system battery charge level goes down to 5%, then the system will be woken up by the
          RTC alarm and hibernated.

**Comment by lobotomm** [comment]
Hi!
I spend endless hours why my Manjaro does not hibernate any more and ended here.
I can only support this issue. Please make both possible at the same time.
HibernateDelaySec and SuspendEstimationSec should work side by side!

Thanks!

**Comment by YHNdnzj** [comment]
> @YHNdnzj do you intend to look at this for this milestone or should I bump it?

I did a quick browse of the code and I believe this should be already supported, yet I have no hardware to test this currently...

@yuwata Can you shed some light on this? Else I'd bump this to the next milestone, and test it when I'm able to.

**Comment by yuwata** [comment]
I neither have one to test the issue. And I have no idea about the current status, as I did not finish the mentioned PR due to my health condition at that time. I will take a look, but let's move to the next milestone.

**Comment by nicegamer7** [comment]
I just tested this, it looks like it's already working as requested in this issue. I think all that's needed is to update the documentation.

Here's what I did to test:
1. Set `HibernateDelaySec` to something high: I used `2h`
2. Discharge battery significantly: I waited till I was at 5%
3. Run `suspend-then-hibernate`: System immediately entered hibernation

Let me know if there's anything else I should test.

**Comment by benklop** [comment]
I think it would also make sense to test while discharging to ~6%, so that the system does enter sleep initally. the expectation would be that the system would wake when the battery reaches 5%, then hibernates.

**Comment by nicegamer7** [comment]
I agree... I actually did try from 6%, but it looks like the battery discharge rate on my laptop while it's in sleep is about 0.5%/hour. So when I tried this with `HibernateDelaySec=2h`, I woke up the laptop after about 1 hour and 57 minutes, and it was not hibernating since the battery was at 5% (I then tried it again and it hibernated immediately).

Now that I know the discharge rate, I'll try this again with something higher, maybe `16h`, and I'll check after 6 hours, since the battery should be below 5% and nowhere near the `16h`.

**Comment by YHNdnzj** [comment]
I actually think that this should be fixed by fbc1dbf29. Could someone try systemd v255?

**Comment by boudekerk** [comment]
I just tested this with systemd 255.2.

```
[Sleep]
HibernateDelaySec=30h
```

Closed lid (suspend-then-hibernate) with 6% battery and then reopened it about 2 hours later. Battery was at 3%, the system was not hibernating.

**Comment by boudekerk** [comment]
@YHNdnzj Is there any update that can be shared?

**Comment by alecbcs** [comment]
I'm also experiencing this behavior with v257. I set `HibernateDelaySec=24h` to have my laptop hibernate after being suspended for 24h but this seems to inhibit the laptop from checking the battery and suspending when < 5%. My laptop (Framework 13) has relatively high battery drain meaning I end up regularly finding my laptop with dead (0%) after leaving it for half a day if it wasn't fully charged when I first put it to sleep.

**Comment by boudekerk** [comment]
@yuwata @YHNdnzj Could you please pick this up in the new year? It's now almost two years old and clearly affecting folks. Thank you for considering.

**Comment by NicholasFlamy** [comment]
It would be good to fix this and update the docs to match.

**Comment by boudekerk** [comment]
@yuwata @YHNdnzj Is there any update? Any chance of this ever being picked back up?
Otherwise perhaps better to close as "won't fix".

**Comment by NicholasFlamy** [comment]
> [@yuwata](https://github.com/yuwata) [@YHNdnzj](https://github.com/YHNdnzj) Is there any update? Any chance of this ever being picked back up? Otherwise perhaps better to close as "won't fix".

If you look at the current milestone that this issue is in, there are 105 open issues. I think they just haven't gotten to this with everything else going on, and I think there are multiple developers not interested in even looking at fixing this.

**Comment by NicholasFlamy** [comment]
Thank you @gvenugo3! Can't wait to test this.
